### PR TITLE
Usage Based Billing

### PR DESCRIPTION
**Pull request by bhelx**
This introduces some new resources and updates to current resources to
support the usage based billing (metered billing) feature.

https://docs.recurly.com/docs/usage-based-billing

Merged Ruby Version: https://github.com/recurly/recurly-client-ruby/pull/237

## Testing

```python
import uuid
import datetime
import time

plan_code = str(uuid.uuid4())

plan = recurly.Plan(plan_code=plan_code, name=plan_code)
plan.unit_amount_in_cents = recurly.Money(USD=1000, EUR=800)
plan.setup_fee_in_cents = recurly.Money(USD=6000, EUR=4500)
plan.plan_interval_length = 1
plan.plan_interval_unit = 'months'
plan.tax_exempt = False
plan.save()

# we only need id but fetching for effect
measured_unit = recurly.MeasuredUnit.get(18783)

addon = recurly.AddOn(
        add_on_code='marketing_emails',
        name='Marketing Emails',
        unit_amount_in_cents=recurly.Money(USD=5, EUR=5),
        add_on_type="usage",
        optional=True,
        usage_type="price",
        measured_unit_id=measured_unit.id,
        )

plan.create_add_on(addon)

subscription = recurly.Subscription()
subscription.plan_code = plan_code

subscription.account = recurly.Account(account_code='usageaccount')

addon1 = recurly.SubscriptionAddOn()
addon1.add_on_code = 'marketing_emails'
addon1.quantity = 1

subscription.subscription_add_ons = [addon1]

subscription.save()

time.sleep(1)

# grabbing the first add_on, will probably want to filter by code
sub_add_on = subscription.subscription_add_ons[0]

usage = recurly.Usage()
usage.amount = 100 # record 100 emails
usage.merchant_tag = "Recording 100 emails used by customer"
usage.recording_timestamp = datetime.datetime.utcnow()
usage.usage_timestamp = datetime.datetime.utcnow()

subscription.create_usage(sub_add_on, usage)

usage = recurly.Usage()
usage.amount = 200 # record 200 emails
usage.merchant_tag = "Recording 200 emails used by customer"
usage.recording_timestamp = datetime.datetime.utcnow()
usage.usage_timestamp = datetime.datetime.utcnow()

subscription.create_usage(sub_add_on, usage)

# prints our 2 usage objects
usages = sub_add_on.usage()

for usage in usages:
    print usage.amount
    print usage.merchant_tag

# You can also filter usages with options
search_options = {
  'billing_status': 'all', # (all or unbilled or billed)
  'datetime_type': 'usage',  # (usage or recording)
}

filtered_usages = sub_add_on.usage(**search_options) # Our usage search options

for usage in filtered_usages:
    print usage
```

### CODE REVIEW

File: recurly/__init__.py [hunk 1]
```diff
@@ -911,11 +951,13 @@ class AddOn(Resource):
...
         'accounting_code',
         'unit_amount_in_cents',
+        'measured_unit_id',
+        'usage_type',
+        'add_on_type',
         'tax_code',
         'created_at',
     )
 
-
```
Comment (bhelx) [inline recurly/__init__.py]: Should probably put this back and put in own commit

**Comment by csmb** [comment]
Looks good to me
